server/src/mysql/worker.rs

```diff
@@ -0,0 +1,129 @@
+// Copyright 2022 CeresDB Project Authors. Licensed under Apache-2.0.
+
+use std::{io, marker::PhantomData, sync::Arc};
+
+use catalog::manager::Manager as CatalogManager;
+use log::{error, info};
+use opensrv_mysql::{AsyncMysqlShim, ErrorKind, QueryResultWriter, StatementMetaWriter};
+use query_engine::executor::Executor as QueryExecutor;
+use snafu::ResultExt;
+use table_engine::engine::EngineRuntimes;
+
+use crate::{
+    context::RequestContext,
+    handlers::{self, sql::Request},
+    instance::Instance,
+    mysql::{error::*, writer::MysqlQueryResultWriter},
+};
+pub struct MysqlWorker<W: std::io::Write + Send + Sync, C, Q> {
+    generic_hold: PhantomData<W>,
+    instance: Arc<Instance<C, Q>>,
+    runtimes: Arc<EngineRuntimes>,
+}
+
+impl<W: std::io::Write + Send + Sync, C: CatalogManager + 'static, Q: QueryExecutor + 'static>
+    MysqlWorker<W, C, Q>
+{
+    pub fn new(instance: Arc<Instance<C, Q>>, runtimes: Arc<EngineRuntimes>) -> Self {
+        Self {
+            generic_hold: PhantomData::default(),
+            instance,
+            runtimes,
+        }
+    }
+}
+
+#[async_trait::async_trait]
+impl<W: std::io::Write + Send + Sync, C: CatalogManager + 'static, Q: QueryExecutor + 'static>
+    AsyncMysqlShim<W> for MysqlWorker<W, C, Q>
+{
+    type Error = std::io::Error;
```

Comment:
I find it the `Error` here requires `From<io::Error>` and I guess it will be more elegant to just `impl From<io::Error> for Error` so that all the returned results is our `Result`.

Ref: https://docs.rs/opensrv-mysql/0.1.0/opensrv_mysql/trait.AsyncMysqlShim.html